sparkle/output/configuration_output.py

```diff
@@ -0,0 +1,256 @@
+            f"{solver_dir_name}_scenario" / "state-run0"
+        if scenario_dir.is_dir():
+            scenario_file = scenario_dir / "scenario.txt"
+            if not scenario_file.is_file():
+                print("Can't find scenario file")
+                return
+        else:
+            print("Can't find scenario file")
+            return
+
+        scenario_dir = scenario_dir.parent
+        # Sets scenario on configurator object
+        self.configurator.scenario = \
+            ConfigurationScenario.from_file(scenario_file, self.solver,
+                                            self.instance_set_train,
+                                            scenario_dir)
+        self.configurator.scenario._set_paths(self.configurator.output_path)
+
+        # Retrieve all configurations
+        self.configurations = self.get_configurations(path)
+
+        # Retrieves validation results and best configuration
+        self.training, self.best_config =\
+            self.get_validation_data(self.instance_set_train)
+
+        # Retrieve test validation results if they exist
+        if self.instance_set_test is not None:
+            self.test, _ = self.get_validation_data(self.instance_set_test)
+
+    def parse_string_to_dict(self: ConfigurationOutput, input_string: str) -> dict:
+        """Convert a configuration string to a dictionary."""
+        result_dict = {}
+        input_string = input_string.replace("'", "")
+        input_string = input_string.replace(",", "")
+        input_string = input_string.replace(":", "")
+        input_string = input_string.replace("{", "")
+        input_string = input_string.replace("}", "")
+        tokens = input_string.split()
+
+        for i in range(0, len(tokens), 2):
+            key = tokens[i]
+            value = tokens[i + 1]
+            result_dict[key] = value
+
+        return result_dict
+
+    def get_configurations(self: ConfigurationOutput, path: Path) -> list[dict]:
+        """Read all configurations and transform them to dictionaries."""
+        config_path = path / "validation" / "configurations.csv"
+        configs = []
+
+        # TODO: Should we check if the configurations are unique?
+        # Check if the path exists and is a file
+        if config_path.exists() and config_path.is_file():
+            with config_path.open("r") as file:
+                for line in file:
+                    config = self.parse_string_to_dict(line.strip())
+                    configs.append(config)
+
+        else:
+            print("Can't find configurations")
+
+        return configs
+
+    def get_validation_data(self: ConfigurationOutput, instance_set: InstanceSet) -> \
+            tuple[ConfigurationResults, str]:
+        """Returns best config and ConfigurationResults for instance set."""
+        objective = self.configurator.scenario.sparkle_objective
+
+        # Retrieve found configuration
+        _, best_config = self.configurator.get_optimal_configuration(
```

Comment:
Wouldn't it be good to save the performance as well (instead of _)?

Reply:
I saw that in generate_report_for_configuration.py it was not being used
I can include it in the report, but can you tell me what kind of performance it is (where in the configuration.json output I should include the variable/what it is called)